sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -313,32 +306,23 @@ class ProjectOverviewView extends VerticalLayout{
 
     /**
      * Determines the state of the current status. Is it in progress or did it complete already
-     * @param samplesInStatus the count for the specific status
-     * @param relativeCount
+     * @param totalCount The total number of samples registered
+     * @param failingSamples The number of failing samples
+     * @param passingSamples The number of passing samples
      */
-    private static State determineCompleteness(RelativeCount relativeCount) {
-        int samplesInStatus = relativeCount.getValue()
-        int totalSamples = relativeCount.getTotal()
-        if (samplesInStatus == totalSamples) {
+    private static State determineCompleteness(int totalCount, int failingSamples, int passingSamples) {
```

Comment:
As a suggestion, you could merge the RelativeCount with the SampleCount so that it contains the tree parameters you provide to this method here. This would make the code a bit leaner and you could get rid of the RelativeCount and provide the number of passed in to `toString`.
The relative passed count could then be changed in the future if failed sample information should be displayed to the user or you want to display the information about a sample status differently. Currently, you would have to adjust both classes in this case. Thus I would say they have the same responsibility in showing the progression of single status to the user and can be merged.
You could then call this method with the status count instead.

Reply:
I very much like your suggestion, I will make the change!